quinn-h3/src/proto/settings.rs

```diff
@@ -0,0 +1,344 @@
+use bytes::{Buf, BufMut};
+use quinn_proto::{
+    coding::{BufExt, BufMutExt, Codec, UnexpectedEnd},
+    VarInt,
+};
+
+use super::frame::{FrameHeader, Type as FrameType};
+
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Hash)]
+pub struct SettingId(pub u64);
+
+macro_rules! setting_identifiers {
+    {$($name:ident = $val:expr,)*} => {
+        impl SettingId {
+            $(pub const $name: SettingId = SettingId($val);)*
+        }
+    }
+}
+
+impl SettingId {
+    const NONE: SettingId = SettingId(0);
+}
+
+setting_identifiers! {
+    QPACK_MAX_TABLE_CAPACITY = 0x1,
+    QPACK_MAX_BLOCKED_STREAMS = 0x7,
+    MAX_HEADER_LIST_SIZE = 0x6,
+}
+
+impl Codec for SettingId {
+    fn decode<B: Buf>(buf: &mut B) -> Result<Self, UnexpectedEnd> {
+        Ok(SettingId(buf.get_var()?))
+    }
+    fn encode<B: BufMut>(&self, buf: &mut B) {
+        buf.write_var(self.0);
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Settings {
+    max_header_list_size: u64,
+    qpack_max_table_capacity: u64,
+    qpack_max_blocked_streams: u64,
+}
+
+impl Default for Settings {
+    /// Create settings with H3 default values
+    fn default() -> Self {
+        Self {
+            max_header_list_size: DEFAULT_PROTO_MAX_HEADER_LIST_SIZE,
+            qpack_max_table_capacity: DEFAULT_PROTO_QPACK_MAX_TABLE_CAPACITY,
+            qpack_max_blocked_streams: DEFAULT_PROTO_QPACK_MAX_BLOCKED_STREAMS,
+        }
+    }
+}
+
+impl Settings {
+    /// Create settings with quinn-h3's recomended values
+    pub fn new() -> Self {
+        Self {
+            max_header_list_size: DEFAULT_MAX_HEADER_LIST_SIZE,
```

Comment:
I think we should just inline the values here, not much point keeping them as constants? You can still use `DEFAULT_MAX_HEADER_LIST_SIZE` to make it clear that that remains at the protocol default value.

Reply:
Got them inlined in a new commit